src/components/checkbox/checkbox.njk

```diff
@@ -1,30 +1,73 @@
-<legend class="rvt-sr-only">{{ legend }}</legend>
-<ul{% if list %} class="rvt-plain-list"{% else %} class="rvt-inline-list"{% endif %}>
-  {% for box in boxes %}
-    <li{% if box.hidden %} class="rvt-checkbox-wrapper"{% endif %}>
-      <input class="rvt-checkbox" type="checkbox" name="checkbox-{{ box.id }}" id="{{ box.id }}"{% if box.modifiers %}{% for modifier in box.modifiers %} {{ modifier.status }}{% endfor %}{% endif %}>
-      {% if box.hidden %}
+<fieldset class="rvt-fieldset">
+  <legend class="rvt-sr-only">Checkbox list</legend>
+  <ul class="rvt-plain-list rvt-width-xl">
+    <li>
+      <span class="rvt-checkbox">
+        <input class="rvt-checkbox__input" type="checkbox" name="checkbox-demo" id="checkbox-1">
+        <!--
+          Can work with hidden input elements that some framework template languages generate
+        -->
         <input type="hidden">
-      {% endif %}
-      <label class="rvt-label" for="{{ box.id }}">{{ box.checkboxLabel }}</label>
-      {% if (box.id === "checkbox-indeterminate") %}
-        <ul class="rvt-plain-list rvt-p-left-lg">
-          {% for item in box.items %}
-            <li>
-              <input class="rvt-checkbox" type="checkbox" name="checkbox-{{ item.id }}" id="{{ item.id }}"{% if loop.index === 1 %} checked{% endif %}>
-              <label class="rvt-label" for="{{ item.id }}">{{ item.checkboxLabel }}</label>
-            </li>
-          {% endfor %}
-        </ul>
-      {% endif %}
+        <label for="checkbox-1">Option one</label>
+      </span>
     </li>
-  {% endfor %}
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="indeterminate">
+        <label for="indeterminate">Option two</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-3" disabled>
+        <label for="checkbox-3">Option three</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-4" disabled checked>
+        <label for="checkbox-4">Option four</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-long">
+        <label for="checkbox-long">
+          <span class="rvt-text-bold">Just a quick note</span><br>
+          <span class="rvt-ts-14 rvt-display-inline-block rvt-color-black-600">This checkbox has a really long label that can wrap on to two lines and still have nice left alignment.</span>
```

Comment:
This is a good example, that I hadn't thought of. But hint text like this shouldn't live in the label. Yes, it increases the hit area, but it results in poor experience for screen reader users. That is essentially one long label. In VoiceOver, that gets read out before saying that it is a checkbox and its checked state ("[label] [secondary text], [checked/unchecked], checkbox"). I've done several iterations of this idea (Options 6 to 9 in the [CodePen](https://codepen.io/basham/pen/yLVmBRJ)). The best result is having this hint text separate from the description and attached to the checkbox by `aria-describedby`. This means that the screen reader announces the hint text after it says it is a checkbox ("[label], [checked/unchecked] checkbox, [description]"), which is a better experience.

We could leave it up to the developer to style this (Option 8), but I think we should provide a `.rvt-checkbox__description` to simplify and standardize it (Option 9).

```html
<div class="rvt-checkbox">
  <input aria-describedby="checkbox-9-description" type="checkbox" name="checkbox-demo" id="checkbox-9">
  <label for="checkbox-9">Option nine</label>
  <div id="checkbox-9-description" class="rvt-checkbox__description">This checkbox has a really long label that can wrap on to two lines and still have nice left alignment.</div>
</div>
```

```css
.rvt-checkbox__description {
  color: var(--rvt-color-black-600);
  font-size: calc(14rem/16);
  padding: 0 0.5rem;
}
```

Reply:
Also, I'm not convinced bolding the label is needed. The style differences with the description text should be sufficient visual contrast.

Reply:
@basham Cool, I like option 9 in the demo. I'll make that update and draft some docs for it in the README in Fractal